File: backend/src/main/java/com/back/back9/domain/user/dto/UserDto.java
package com.back.back9.domain.user.dto;

import com.back.back9.domain.user.entity.User;

import java.time.LocalDateTime;

public record UserDto(
        Long id,
        String userLoginId,
        String username,
        String role,
        LocalDateTime createdAt,
        LocalDateTime modifiedAt
) {
    public UserDto(Long id, String userLoginId, String username, String role, LocalDateTime createdAt, LocalDateTime modifiedAt) {
        this.id = id;
        this.userLoginId = userLoginId;
        this.username = username;
        this.role = role;
        this.createdAt = createdAt;
        this.modifiedAt = modifiedAt;
    }

    public UserDto(User user) {
        this(
                user.getId(),
                user.getUserLoginId(),
                user.getUsername(),
                user.getRole().name(),
                user.getCreatedAt(),
                user.getModifiedAt()
        );
    }
}

File: backend/src/main/kotlin/com/back/back9/domain/user/dto/UserDto.kt
package com.back.back9.domain.user.dto

import com.back.back9.domain.user.entity.User
import java.time.LocalDateTime

data class UserDto(
        val id: Long?,
        val userLoginId: String,
        val username: String,
        val role: String,
        val createdAt: LocalDateTime?,
        val modifiedAt: LocalDateTime?
) {
    companion object {
        fun from(user: User): UserDto {
            return UserDto(
                id = user.id,
                userLoginId = user.userLoginId,
                username = user.username,
                role = user.role.name,
                createdAt = user.createdAt,
                modifiedAt = user.modifiedAt
            )
        }
    }
}